src/charm.py

```diff
@@ -424,6 +453,83 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
                 f"Migration of sharding components not permitted, revert config role to {self.role}"
             )
 
+    def __fix_users_for_reuse(self, direct_mongo: MongoDBConnection) -> None:
+        """Fix the users in the DB for storage reuse."""
+        users = direct_mongo.get_all_users()
```

Comment:
This call is controlled here since we use it only to know which users are already present in the database.
We're only changing/dropping the administrative users that are already in DB (at most `operator`, `backup` and `monitor`)